ethcore/light/src/on_demand/mod.rs

```diff
@@ -326,30 +351,68 @@ pub struct OnDemand {
 	in_transit: RwLock<HashMap<ReqId, Pending>>,
 	cache: Arc<Mutex<Cache>>,
 	no_immediate_dispatch: bool,
-	base_retry_count: usize,
-	query_inactive_time_limit: Option<Duration>,
+	response_time_window: Duration,
+	request_time_window: Duration,
+	request_backoff_start: Duration,
+	request_backoff_max: Duration,
+	request_backoff_rounds_max: usize
 }
 
 impl OnDemand {
 
 	/// Create a new `OnDemand` service with the given cache.
-	pub fn new(cache: Arc<Mutex<Cache>>) -> Self {
+	pub fn new(
+		cache: Arc<Mutex<Cache>>,
+		response_time_window: Duration,
+		request_time_window: Duration,
+		request_backoff_start: Duration,
+		request_backoff_max: Duration,
+		request_backoff_rounds_max: usize,
+	) -> Self {
+
 		OnDemand {
 			pending: RwLock::new(Vec::new()),
 			peers: RwLock::new(HashMap::new()),
 			in_transit: RwLock::new(HashMap::new()),
 			cache,
 			no_immediate_dispatch: false,
-			base_retry_count: DEFAULT_RETRY_COUNT,
-			query_inactive_time_limit: Some(DEFAULT_QUERY_TIME_LIMIT),
+			response_time_window: Self::santize_circuit_breaker_input(response_time_window, "Response time window"),
+			request_time_window: Self::santize_circuit_breaker_input(request_time_window, "Request time window"),
+			request_backoff_start: Self::santize_circuit_breaker_input(request_backoff_start, "Request initial backoff time window"),
+			request_backoff_max: Self::santize_circuit_breaker_input(request_backoff_max, "Request maximum backoff time window"),
+			request_backoff_rounds_max,
+		}
+	}
+
+	fn santize_circuit_breaker_input(dur: Duration, name: &'static str) -> Duration {
+		if dur.as_secs() < 1 {
+			warn!(target: "on_demand",
+				"{} is too short must be at least 1 second, configuring it to 1 second", name);
```

Comment:
If it's not allowed to be less than 1 second, maybe we should use seconds instead of ms for duration params?